lib/Autocomplete.js

```diff
@@ -375,7 +375,7 @@ class Autocomplete extends React.Component {
       const itemValue = getItemValue(matchedItem)
       const itemValueDoesMatch = (itemValue.toLowerCase().indexOf(
         value.toLowerCase()
-      ) === 0)
+      ) > -1)
```

Comment:
I'd argue, that `itemValueDoesMatch` is unnecessary altogether.
If an item is displayed because it passed the check of `shouldItemRender` then we can also auto-highlight it if it's the first item in the list.